portal/static/js/main.js

```diff
@@ -5550,6 +5621,31 @@ var tnthAjax = {
              if (callback) callback({"error": i18next.t("no data returned")});
             };
         });
+    },
+    "setConfigurationUI": function(configKey, value) {
+        if (configKey) {
+            if ($("#profile_" + configKey).length === 0) {
+                $("body").append("<input type='hidden' id='profile_" + configKey + "' value='" + (value?value:"") + "'/>");
+            }
+        }
+    },
+    "getConfiguration": function(configVar, userId, params, callback, setConfigInUI) {
+        callback = callback || function() {};
+        if (!configVar) {
+            callback({"error": "a configuration variable name is required"});
+            return false;
+        }
+        var self = this;
+        this.sendRequest("/api/settings/"+ configVar, "GET", userId, (params||{}), function(data) {
```

Comment:
Would you be able to cache the config settings in localStorage (and only make the request if necessary)?